presto-native-execution/presto_cpp/main/common/Configs.cpp

```diff
@@ -718,14 +732,34 @@ BaseVeloxQueryConfig* BaseVeloxQueryConfig::instance() {
   return instance.get();
 }
 
-void BaseVeloxQueryConfig::initialize(const std::string& filePath) {
-  ConfigBase::initialize(filePath);
-  update(*SystemConfig::instance());
-}
+void BaseVeloxQueryConfig::updateLoadedValues(
+    std::unordered_map<std::string, std::string>& values) const {
+  /// Update velox config with values from presto system config.
+  auto systemConfig = SystemConfig::instance();
+
+  using namespace velox::core;
+  const std::unordered_map<std::string, std::string> updatedValues{
+      {QueryConfig::kPrestoArrayAggIgnoreNulls,
+       bool2String(systemConfig->useLegacyArrayAgg())},
+      {QueryConfig::kMaxArbitraryBufferSize,
+       systemConfig->capacityPropertyAsBytesString(
```

Comment:
I wonder if Velox can parse these strings properly.

Reply:
@mbasmanova 
I don't think Velox has the functionailty at the moment.
We can add it later and change this code to use it.